bigquery_etl/view/publish_views.py

```diff
@@ -1,184 +0,0 @@
-
-import click
-import sqlparse
-from google.api_core.exceptions import BadRequest
-from google.cloud import bigquery
-
-from bigquery_etl.schema import Schema
-
-VIEWS_TO_SKIP = (
-    # Access Denied
-    "activity_stream/tile_id_types/view.sql",
-    "pocket/pocket_reach_mau/view.sql",
-    "telemetry/buildhub2/view.sql",
-    # Dataset glam-fenix-dev:glam_etl was not found
-    # TODO: this should be removed if views are to be automatically deployed
-    *[str(path) for path in Path("sql/glam-fenix-dev").glob("glam_etl/**/view.sql")],
-)
-
-# suffixes of datasets with non-user-facing views
-NON_USER_FACING_DATASET_SUFFIXES = (
-    "_derived",
-    "_external",
-    "_bi",
-    "_restricted",
-)
-
-
-def _process_file(client, args, filepath):
-    if any(filepath.endswith(p) for p in VIEWS_TO_SKIP):
-        print(f"Skipping authorized view definition {filepath}")
-        return True
-    with open(filepath) as f:
-        sql = f.read()
-    parsed = sqlparse.parse(sql)[0]
-    tokens = [
-        t
-        for t in parsed.tokens
-        if not (t.is_whitespace or isinstance(t, sqlparse.sql.Comment))
-    ]
-    is_view_statement = (
-        " ".join(tokens[0].normalized.split()) == "CREATE OR REPLACE"
-        and tokens[1].normalized == "VIEW"
-    )
-    if is_view_statement:
-        target_view_orig = str(tokens[2]).strip().split()[0]
-        target_view = target_view_orig
-        if args.target_project:
-            # target_view must be a fully-qualified BigQuery Standard SQL table
-            # identifier, which is of the form f"{project_id}.{dataset_id}.{table_id}".
-            # dataset_id and table_id may not contain "." or "`". Each component may be
-            # a backtick (`) quoted identifier, or the whole thing may be a backtick
-            # quoted identifier, but not both.
-            # Project IDs must contain 6-63 lowercase letters, digits, or dashes. Some
-            # project IDs also include domain name separated by a colon. IDs must start
-            # with a letter and may not end with a dash. For more information see also
-            # https://github.com/mozilla/bigquery-etl/pull/1427#issuecomment-707376291
-            project_id = target_view_orig.replace("`", "").rsplit(".", 2)[0]
-            # Only views for moz-fx-data-shared-prod will get published to other
-            # projects, for example to moz-fx-data-derived-datasets
-            if project_id != "moz-fx-data-shared-prod":
-                print(f"Skipping {filepath} because --target-project is set")
-                return True
-            target_view = target_view_orig.replace(project_id, args.target_project, 1)
-            # We only change the first occurrence, which is in the target view name.
-            sql = sql.replace(project_id, args.target_project, 1)
-        job_config = bigquery.QueryJobConfig(use_legacy_sql=False, dry_run=args.dry_run)
-        query_job = client.query(sql, job_config)
```

Comment:
This validation code is almost identical to the logic in the old `validate_views.py` file. Moving both into a `View` class makes it easier to reuse the validation logic.